File: Java/0167-two-sum-ii-input-array-is-sorted.java
import javax.print.attribute.standard.NumberUpSupported;

class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int left = 0, right = numbers.length - 1;
        //int[] res = new int[numbers.length];
        while(left < right){
            int s = numbers[left] + numbers[right];
            if(s == target){
                return new int[]{left+1, right+1};
            }
            else if( s < target)
                left++;
            else
                right--;
        }
        return new int[0];
    }
}

File: C++/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int left = 0, right  = numbers.size() - 1;
        vector<int> res;
        while(left < right){
            int s = numbers[left] + numbers[right];
            if(s == target){
                res.push_back(left+1);
                res.push_back(right+1);
                return res;
            }
            else if(s < target)
                left++;
            else
                right--;
        }
        return {0};
    }
};